tests/manage/pv_services/pv_encryption/test_pvc_with_tc_overrie_vn.py

```diff
@@ -0,0 +1,178 @@
+if config.ENV_DATA.get("vault_hcp"):
+    argvalues = [
+        pytest.param(
+            "v1", kmsprovider, True, marks=pytest.mark.polarion_id("OCS-4639")
+        ),
+        pytest.param(
+            "v2", kmsprovider, True, marks=pytest.mark.polarion_id("OCS-4641")
+        ),
+    ]
+else:
+    argvalues = [
+        pytest.param(
+            "v1", kmsprovider, False, marks=pytest.mark.polarion_id("OCS-4638")
+        ),
+        pytest.param(
+            "v2", kmsprovider, False, marks=pytest.mark.polarion_id("OCS-4640")
+        ),
+    ]
+
+
+@pytest.mark.parametrize(
+    argnames=argnames,
+    argvalues=argvalues,
+)
+@tier1
+@skipif_ocs_version("<4.8")
+@bugzilla("2050056")
+class TestEncryptedRbdTenantConfigmapOverride(ManageTest):
+    """
+    Tests to check Tenant configmap override vault namespace or not
+    1. Create a new namespace in OCP and create the ceph-csi-kms-token secret
+    2. Create an encryption enabled storageclass using vault namespaces
+    3. Create a configmap in the tenant namespace to override the vault namespace
+    4. Update the ceph-csi-kms-token secret in the tenant namespace to provide access to the new namespace and backend
+       path
+    5. Create a new PVC
+    6. Verify that the PVC is bound and the key is created in the specified path in Vault
+    7. Mounting the PVC in an app pod and running IO
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        kv_version,
+        kms_provider,
+        use_vault_namespace,
+        pv_encryption_kms_setup_factory,
+        project_factory,
+        storageclass_factory,
+        pod_factory,
+    ):
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+
+        log.info("Setting up csi-kms-connection-details configmap")
+        self.kms = pv_encryption_kms_setup_factory(kv_version, use_vault_namespace)
+        log.info("csi-kms-connection-details setup successful")
+
+        # Create a project
+        self.proj_obj = project_factory()
+
+        # Create an encryption enabled storage class for RBD
+        self.sc_obj = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=self.kms.kmsid,
+        )
+
+    def test_encryptedrbd_pvc_status_with_tenant_configmap_override(
+        self, multi_pvc_factory, kv_version, kms_provider, pod_factory
+    ):
+
+        # Create new backend-path
+        vault_resource_name = create_unique_resource_name("test", "vault")
+        self.kms.vault_create_backend_path(
+            backend_path=vault_resource_name, kv_version=kv_version
+        )
+
+        # Create policy
+        self.kms.vault_create_policy(policy_name=vault_resource_name)
+
+        # Create a configmap in the tenant namespace to override the vault namespace as shown below:
+        self.kms.create_tenant_configmap(
+            self.proj_obj.namespace, vaultBackendPath=f"{vault_resource_name}"
```

Comment:
This is only overriding the backend path. The bug is for overriding the vault namespace.